lib/msprime.c

```diff
@@ -3945,6 +3949,10 @@ msp_add_simple_bottleneck(msp_t *self, double time, int population_id, double pr
         ret = MSP_ERR_BAD_PARAM_VALUE;
         goto out;
     }
+    if (self->model.type == MSP_MODEL_DTWF) {
```

Comment:
I don't think there's much point in catching these at 'add' time, as often we'll be using a different model at the start and end of the simulation. Might as well just throw the error at run time so we have less code to test. (Same for the other bottleneck below).